pallets/thea/src/lib.rs

```diff
@@ -481,6 +510,42 @@ pub mod pallet {
 			));
 			Ok(())
 		}
+
+		/// Extrinsic to acknowledge on chain state key change completion on all foreign chains
+		///
+		/// # Parameters
+		///
+		/// * `origin`: Any relayer
+		/// * `network`: Network id
+		/// * `tx_hash`: Transaction hash of key update on foreign chain
+		/// * `bit_map`: Bitmap of Thea relayers
+		/// * `bls_signature`: BLS signature of relayers
+		// TODO: [Issue #606] Use benchmarks
+		#[pallet::weight(1000)]
+		pub fn thea_key_rotation_complete(
+			origin: OriginFor<T>,
+			network: Network,
+			tx_hash: sp_core::H256,
+			bit_map: u128,
+			bls_signature: [u8; 96],
+		) -> DispatchResult {
+			let _relayer = ensure_signed(origin)?;
+			// Fetch current active relayer set BLS Keys
+			let current_active_relayer_set = Self::get_relayers_key_vector(network);
+
+			// Call host function with current_active_relayer_set, signature, bit_map, verify nonce
+			ensure!(
+				thea_primitives::thea_ext::bls_verify(
+					&bls_signature,
+					bit_map,
+					&tx_hash.encode(),
+					&current_active_relayer_set
+				),
+				Error::<T>::BLSSignatureVerificationFailed
+			);
```

Comment:
We should track this hash, as there exist a scenario that can lead to replay attack if relayers remain unchanged over different sets .